### PR TITLE
M2-5742: Dataviz - Number Selection Item support

### PR DESCRIPTION
**Pull request by alex-i-e**
- [x] Tests for the changes have been added
- [X] Related documentation has been added / updated
- [x] Delivered the `fix` or `feature` branches into `develop` or `release` branches via `Squash and Merge` (to keep clean history)

### 📝 Description

🔗 [Jira Ticket M2-5742](https://mindlogger.atlassian.net/browse/M2-5742)

### 📸 Screenshots

| Before (Optional)                      | After                                 |
| -------------------------------------- | ------------------------------------- |
| image | image |
| image | image |

### CODE REVIEW

File: src/modules/Dashboard/features/RespondentData/RespondentDataSummary/Report/ResponseOptions/ResponseOptions.utils.tsx [hunk 1]
```diff
@@ -56,8 +56,27 @@ export const getResponseItem = ({
...
     case ItemResponseType.SingleSelection:
     case ItemResponseType.MultipleSelection:
-    case ItemResponseType.Slider:
-      return renderMultipleSelection();
+    case ItemResponseType.Slider: {
+      const { options } = activityItem.responseValues;
+
+      return renderMultipleSelection(options);
+    }
+    case ItemResponseType.NumberSelection: {
```
Comment (IhorNesterenko99) [inline src/modules/Dashboard/features/RespondentData/RespondentDataSummary/Report/ResponseOptions/ResponseOptions.utils.tsx:64]: Please add tests to cover changes to the ResponseOptions.test.tsx

File: src/modules/Dashboard/features/RespondentData/RespondentDataSummary/Report/Report.utils.ts [hunk 2]
```diff
@@ -452,6 +453,20 @@ export const formatActivityItemAnswers = (
         answers,
       };
     }
+    case ItemResponseType.NumberSelection: {
```
Comment (IhorNesterenko99) [inline src/modules/Dashboard/features/RespondentData/RespondentDataSummary/Report/Report.utils.ts:456]: Please add this case to the Report.utils.test.ts

File: src/modules/Dashboard/features/RespondentData/RespondentDataReview/Review/Review.utils.tsx [hunk 3]
```diff
@@ -67,6 +69,8 @@ export const getResponseItem = (activityItemAnswer: ActivityItemAnswer) => {
       return <SliderResponseItem {...(activityItemAnswer as SliderItemAnswer)} />;
     case ItemResponseType.Text:
       return <TextResponseItem {...(activityItemAnswer as TextItemAnswer)} />;
+    case ItemResponseType.NumberSelection:
```
Comment (IhorNesterenko99) [inline src/modules/Dashboard/features/RespondentData/RespondentDataReview/Review/Review.utils.tsx:72]: Please add this case to the Review.utils.test.tsx

File: src/modules/Dashboard/features/RespondentData/RespondentDataReview/NumberSelectionResponseItem/NumberSelectionResponseItem.test.tsx [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+describe('NumberSelectionResponseItem', () => {
+  test.each`
+    answer                  | expected
+    ${{ value: '123' }}     | ${'123'}
+    ${{ value: 123 }}       | ${'123'}
+    ${{ value: 0 }}         | ${'0'}
+    ${{ value: null }}      | ${''}
+    ${{ value: undefined }} | ${''}
+  `('render text response item for answer = $answer', ({ answer, expected }) => {
+    renderWithProviders(<NumberSelectionResponseItem answer={answer} data-testid={dataTestid} />);
```
Comment (IhorNesterenko99) [inline src/modules/Dashboard/features/RespondentData/RespondentDataReview/NumberSelectionResponseItem/NumberSelectionResponseItem.test.tsx:20]: What about using ``getByTestId`` like this: ``const { getByTestId } = renderWithProviders...``?
Or is it better to use ``getByTestId`` as the screen property? What do you think?

Comment (alex-i-e) [reply src/modules/Dashboard/features/RespondentData/RespondentDataReview/NumberSelectionResponseItem/NumberSelectionResponseItem.test.tsx:20]: Here I would agree with Kent C. Dodds 
https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#not-using-screen

**Review by IhorNesterenko99 — APPROVED** [review]

**Pull request merged**